f8a_report/v2/report_generator.py

```diff
@@ -0,0 +1,366 @@
+                    analysed_dependencies)
+
+                self.all_deps[ecosystem].append(stack_info_template['stack'])
+                stack_str = ','.join(stack_info_template['stack'])
+                self.stacks_list[ecosystem].append(stack_str)
+
+                stack_info_template['unknown_dependencies'] = normalised_unknown_dependencies
+                self.all_unknown_deps[ecosystem].append(normalised_unknown_dependencies)
+
+                stack_info_template['license']['unknown'] = unknown_licenses
+                self.all_unknown_lic.append(stack_info_template['license']['unknown'])
+
+                # Accumulating security information.
+                for package in analysed_dependencies:
+
+                    for cve in package.get('public_vulnerabilities'):
+                        stack_info_template['public_vulnerabilities']['cve_list'].append(cve)
+                        self.all_cve_list.append('{cve}:{cvss}'.
+                                                 format(cve=cve['CVE'], cvss=cve['CVSS']))
+
+                    for cve in package.get('private_vulnerabilities'):
+                        stack_info_template['private_vulnerabilities']['cve_list'].append(cve)
+                        self.all_cve_list.append('{cve}:{cvss}'.
+                                                 format(cve=cve['CVE'], cvss=cve['CVSS']))
+
+                ended_at, started_at = self.get_audit_timelines(stack)
+
+                response_time = self.report_helper.datediff_in_millisecs(started_at, ended_at)
+                stack_info_template['response_time'] = '%f ms' % response_time
+                self.total_response_time['all'] += response_time
+                self.total_response_time[stack_info_template['ecosystem']] += response_time
+                report_template['stacks_details'].append(stack_info_template)
+            except (IndexError, KeyError, TypeError) as e:
+                logger.exception('Error: %r' % e)
+                continue
+        logger.info("Stacks Analyse Completed.")
+        return report_template
+
+    def build_report_summary(self, unknown_deps_ingestion_report, report_content) -> dict:
+        """Build Final Report Summary."""
+        logger.info("Building Report summary.")
+
+        total_stack_requests = self.total_stack_requests
+        all_deps = self.all_deps
+        all_unknown_deps = self.all_unknown_deps
+        unique_stacks_with_recurrence_count = self.unique_stacks_with_recurrence_count
+        unique_stacks_with_deps_count = self.unique_stacks_with_deps_count
+        avg_response_time = self.avg_response_time
+        unknown_licenses = self.unknown_licenses
+        all_cve_list = self.all_cve_list
+        total_response_time = self.total_response_time
+        start_date = self.start_date
+        return {
+            'total_stack_requests_count': total_stack_requests['all'],
+            'npm': self.report_helper.get_ecosystem_summary('npm', total_stack_requests,
```

Comment:
Use list comprehensions for these 3 method calls and you do not have to make the call 3 times. That will help in not defining local variables.

[func(stack_name) for stack_name in ['npm', 'maven', 'pypi']]

Reply:
Ack!